pbf/views.py

```diff
@@ -237,13 +258,11 @@ def add_player(request, game_id):
     if '-' in spirit_name:
         spirit_name, aspect = spirit_name.split(' - ')
     spirit = get_object_or_404(Spirit, name=spirit_name)
-    
-    if aspect in spirit_starting_energy.keys():
-        starting_energy = spirit_starting_energy[aspect]
-    else:
-        starting_energy = spirit_starting_energy[spirit.name]
-
-    gp = GamePlayer(game=game, spirit=spirit, color=colors[0], aspect=aspect, starting_energy=starting_energy)
+    spirit_and_aspect = spirit_name + ("/" + aspect if aspect else "")
```

Comment:
it's not lost on me I have added code that does something a little silly: I'm recomputing `spirit_and_aspect` from `spirit_name` and `aspect`... but above, there's already an existing variable that contains both spirit and aspect! It's `request.POST['spirit']` (which might contain `-`). The existing code already splits it apart into `spirit_name` and `aspect`, so it's a fair criticism that this code added by me is doing duplicate work by putting them back together instead of just saving `request.POST['spirit']`.

I think the argument in favour of doing it the way it's done here is that all changes are in a contiguous series of instead of interspersed with existing code, but this argument only benefits short-term reviewability rather than long-term maintainability.

I do have some interest in changing this but right now I manually test each change and it'd be so great to be able to automate that.

Reply:
I've corrected it to use the way that makes more sense. I added tests to ease my process.

Reply:
I will put the tests in a separate pull request since there are separate matters to be discussed regarding the tests.